client/src/lib/slate-plugins/linkifyPlugin.tsx

```diff
@@ -0,0 +1,67 @@
+// Adapted from this https://github.com/ianstormtaylor/slate/blob/3528bb7366f5b87a14fe9adef65a42d2e0eb712f/examples/links/index.js
+import isUrl = require("is-url");
+import * as React from "react";
+import { getEventTransfer } from "slate-react";
+
+function wrapLink(change: any, href: any) {
+  change.insertInline({
+    type: "link",
+    data: { href },
+    nodes: [
+      {
+        object: "text",
+        leaves: [
+          {
+            object: "leaf",
+            text: href,
+          }
+        ]
+      }
+    ]
+  });
+
+  change.collapseToEnd();
+}
+
+function LinkifyPlugin() {
+  return {
+    onPaste: (event, change) => {
+      const transfer = getEventTransfer(event);
+      const { type, text } = transfer;
+      if (type !== "text" && type !== "html") {
```

Comment:
```suggestion
      if (type !== "text" && type !== "html" && !isUrl(text)) {
```

Reply:
The above change would include deleting the bottom segment.

Reply:
I suggest adding some naming to this bit. It's not obvious to me why these conditions work this way.